packages/gestalt/src/IconButtonFloating.js

```diff
@@ -0,0 +1,90 @@
+// @flow strict
+import { forwardRef, type Node } from 'react';
+import Box from './Box.js';
+import IconButton from './IconButton.js';
+import icons from './icons/index.js';
+
+type Props = {|
+  /**
+   * Specifies the `id` of an associated element (or elements) whose contents or visibility are controlled by IconButtonFLoating so that screen reader users can identify the relationship between elements. See the [Accessibility](#ARIA-attributes) guidelines for details on proper usage.
+   */
+  accessibilityControls?: string,
```

Comment:
accessibilityControls should only required with Dropdowns right, not necessarily Modals?

If so, then ok if optional. We should explain this in Docs. 

If not, if it controls Modal too, then it should be required.